modules/core/02-client/v2/keeper/grpc_query.go

```diff
@@ -0,0 +1,40 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/v9/modules/core/02-client/v2/types"
+)
+
+var _ types.QueryServer = (*queryServer)(nil)
+
+// queryServer implements the 02-client/v2 types.QueryServer interface.
+// It embeds the client keeper to leverage store access while limiting the api of the client keeper.
+type queryServer struct {
+	*Keeper
+}
+
+// NewQueryServer returns a new 02-client/v2 types.QueryServer implementation.
+func NewQueryServer(k *Keeper) types.QueryServer {
+	return &queryServer{
+		Keeper: k,
+	}
+}
+
+// CounterpartyInfo gets the CounterpartyInfo from the store corresponding to the request client ID.
+func (q queryServer) CounterpartyInfo(ctx context.Context, request *types.QueryCounterpartyInfoRequest) (*types.QueryCounterpartyInfoResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
```

Comment:
We generally do a bit more validation before the grpc directly gets from store. Would be good to follow the same.

See the v1 chanKeeper grpc implementation